### PR TITLE
Added OrgURLs in customDimension to have better Vscode Desktop Usage count and Copilot Telemetry Improvements

### PR DESCRIPTION
**Pull request by BidishaMS**

### CODE REVIEW

File: src/client/extension.ts [hunk 1]
```diff
@@ -162,6 +162,13 @@ export async function activate(
             (fl) => ({ ...fl, uri: fl.uri.fsPath } as WorkspaceFolder)
         ) || [];
     
+    const data = getPortalsOrgURLs(workspaceFolders);
+    // Added this loop to get all the orgURLs customers are working on. Combination of orgURLs and vscodemachineId help us determine desktop usage
+    data?.forEach(value =>{
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion,  @typescript-eslint/no-extra-non-null-assertion
+        const orgId = value.toLowerCase().match(/([a-z,A-Z,0-9]*.[a-z,A-Z,0-9]*.dynamics.com)/g)![0] // extract the orgURLs from the manifest file name.
```
Comment (tehcrashxor) [inline src/client/extension.ts:169]: Are Portals available for Sovereign clouds?  If so, this regex will miss the *.crm.dynamics.cn for China, *.crm.microsoftdynamics.us and *.crm.appsplatform.us for UsGov

File: src/client/extension.ts [hunk 2]
```diff
@@ -162,6 +162,13 @@ export async function activate(
             (fl) => ({ ...fl, uri: fl.uri.fsPath } as WorkspaceFolder)
         ) || [];
     
+    const data = getPortalsOrgURLs(workspaceFolders);
```
Comment (amitjoshi438) [inline src/client/extension.ts:165]: Will this also handle any update to workspace?

Comment (BidishaMS) [reply src/client/extension.ts:165]: On every change in our workspace folder, this code path is getting triggered

File: src/client/extension.ts (hunk 1 above)
Comment (BidishaMS) [reply src/client/extension.ts]: thanks Crash for pointing this. Fixed

**Review by amitjoshi438 — APPROVED** [review]

File: src/common/copilot/IntelligenceApiService.ts [hunk 3]
```diff
@@ -72,7 +72,7 @@ export async function sendApiRequest(userPrompt: string, activeFileParams: IActi
         const jsonResponse = await response.json();
 
         if (jsonResponse.operationStatus === 'Success') {
-          sendTelemetryEvent(telemetry, { eventName: CopilotResponseSuccessEvent, copilotSessionId: sessionID, durationInMills: responseTime });
+          sendTelemetryEvent(telemetry, { eventName: CopilotResponseSuccessEvent, copilotSessionId: sessionID, durationInMills: responseTime, orgId: orgID });
```
Comment (petrochuk) [inline src/common/copilot/IntelligenceApiService.ts:75]: Where is this OrgId going? Is it properly marked as PII data?

File: src/common/copilot/telemetry/ITelemetry.ts [hunk 4]
```diff
@@ -15,4 +15,7 @@ export interface IProDevCopilotTelemetryData {
     feedbackType?: string,
     responseStatus?: number,
     codeLineCount?: number,
+    geoName?: string,
+    aibEndpoint?: string,
```
Comment (petrochuk) [inline src/common/copilot/telemetry/ITelemetry.ts:19]: Why do we need to send AI Builder endpoint to telemetry?

File: src/common/copilot/user-feedback/CESSurvey.ts [hunk 5]
```diff
@@ -87,7 +87,7 @@ function initializeFeedbackData(sessionId: string): IFeedbackData {
       },
       {
         key: 'scenario',
-        value: 'ProdDevCopilot'
+        value: 'ProDevCopilot'
```
Comment (petrochuk) [inline src/common/copilot/user-feedback/CESSurvey.ts:90]: Should it be Power Pages Copilot?

File: src/common/copilot/telemetry/ITelemetry.ts (hunk 4 above)
Comment (BidishaMS) [reply src/common/copilot/telemetry/ITelemetry.ts:19]: Is it going to the correct aibEndpoint and which island it is going we need for debugging.

File: src/common/copilot/IntelligenceApiService.ts (hunk 3 above)
Comment (petrochuk) [reply src/common/copilot/IntelligenceApiService.ts:75]: Organization Identifiable Information (OII) - still needs to be properly handled

File: src/common/copilot/telemetry/ITelemetry.ts (hunk 4 above)
Comment (petrochuk) [reply src/common/copilot/telemetry/ITelemetry.ts:19]: Can this be part of log files instead?

Comment (BidishaMS) [reply src/common/copilot/telemetry/ITelemetry.ts:19]: Is there any specific reason you think this is a problem? If we put this in log file then we are dependent on Customer to share this for debugging.

File: src/common/copilot/IntelligenceApiService.ts (hunk 3 above)
Comment (BidishaMS) [reply src/common/copilot/IntelligenceApiService.ts:75]: Even today orgId is being logged by **AuthProfile** in customDimension. 

image

**Review by petrochuk — APPROVED** [review]

**Pull request merged**